go/internal/feast/onlinestore/cassandraonlinestore.go

```diff
@@ -0,0 +1,299 @@
+	if !ok {
+		username = "cassandra"
+		log.Warn().Msg("username not defined: Using default username instead")
+	}
+
+	password, ok := onlineStoreConfig["password"]
+	if !ok {
+		password = "cassandra"
+		log.Warn().Msg("password not defined: Using default password instead")
+	}
+
+	var usernameStr string
+	if usernameStr, ok = username.(string); !ok {
+		return nil, fmt.Errorf("failed to convert username to string: %+v", usernameStr)
+	}
+
+	var passwordStr string
+	if passwordStr, ok = password.(string); !ok {
+		return nil, fmt.Errorf("failed to convert password to string: %+v", passwordStr)
+	}
+
+	keyspace, ok := onlineStoreConfig["keyspace"]
+	if !ok {
+		keyspace = "scylladb"
+		log.Warn().Msg("Keyspace not defined: Using 'scylladb' as keyspace instead")
+	}
+	store.keyspace = keyspace.(string)
+
+	var keyspaceStr string
+	if keyspaceStr, ok = keyspace.(string); !ok {
+		return nil, fmt.Errorf("failed to convert keyspace to string: %+v", keyspaceStr)
+	}
+
+	protocolVersion, ok := onlineStoreConfig["protocol_version"]
+	if !ok {
+		protocolVersion = 4.0
+		log.Warn().Msg("protocol_version not specified: Using 4 instead")
+	}
+	protocolVersionInt := int(protocolVersion.(float64))
+
+	redisTraceServiceName := os.Getenv("DD_SERVICE") + "-cassandra"
+	if redisTraceServiceName == "" {
+		redisTraceServiceName = "cassandra.client" // default service name if DD_SERVICE is not set
+	}
+	store.clusterConfigs = gocqltrace.NewCluster(cassandraHostsStr, gocqltrace.WithServiceName(redisTraceServiceName))
+	// TODO: Figure out if we need to offer users the ability to tune the timeouts
+	//store.clusterConfigs.ConnectTimeout = 1
+	//store.clusterConfigs.Timeout = 1
+	store.clusterConfigs.ProtoVersion = protocolVersionInt
+	//store.clusterConfigs.Consistency = gocql.Quorum
+	store.clusterConfigs.Keyspace = keyspaceStr
+	loadBalancingPolicy, ok := onlineStoreConfig["load_balancing"]
+	if !ok {
+		loadBalancingPolicy = gocql.RoundRobinHostPolicy()
+		log.Warn().Msg("No load balancing policy selected; setting Round Robin Host Policy")
+	}
+	loadBalancingPolicyStr, _ := loadBalancingPolicy.(string)
+	if loadBalancingPolicyStr == "DCAwareRoundRobinPolicy" {
+		store.clusterConfigs.PoolConfig.HostSelectionPolicy = gocql.RoundRobinHostPolicy()
+	} else if loadBalancingPolicyStr == "TokenAwarePolicy(DCAwareRoundRobinPolicy)" {
+		// Configure fallback policy if unable to reach the shard
+		fallback := gocql.RoundRobinHostPolicy()
+		// If using ScyllaDB and setting this policy, this makes the driver shard aware to improve performance
+		store.clusterConfigs.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(fallback)
+		if config.OnlineStore["type"] == "scylladb" {
+			store.clusterConfigs.Port = 19042
+		} else {
+			store.clusterConfigs.Port = 9042
+		}
+	}
+
+	store.clusterConfigs.Authenticator = gocql.PasswordAuthenticator{Username: usernameStr, Password: passwordStr}
+	createdSession, err := store.clusterConfigs.CreateSession()
```

Comment:
Is the communication between client and scylladb cluster secure?

Reply:
This also requires config changes in the Scylla clusters (off by default)